keylime/keylime_agent.py

```diff
@@ -745,12 +754,21 @@ def main():
         keylime_ca = os.path.join(config.WORK_DIR, 'cv_ca', 'cacert.crt')
 
     server = CloudAgentHTTPServer(serveraddr, Handler, agent_uuid, contact_ip, ima_log_file, tpm_log_file_data)
-    context = web_util.generate_mtls_context(server.mtls_cert_path, server.rsakey_path, keylime_ca, logger=logger)
-    server.socket = context.wrap_socket(server.socket, server_side=True)
+    if server.mtls_cert_enabled:
+        context = web_util.generate_mtls_context(server.mtls_cert_path, server.rsakey_path, keylime_ca, logger=logger)
+        server.socket = context.wrap_socket(server.socket, server_side=True)
+    else:
+        if not config.getboolean('cloud_agent', 'enable_insecure_payload', fallback=False):
```

Comment:
Instead of not starting, would just not accept payloads by immediately returning instead of handling U/V key and payload POST request.

Somewhere here:
https://github.com/keylime/keylime/blob/99f444dc46861e9dc9eadaadd5267f5e14164f3c/keylime/keylime_agent.py#L210-L230


Adding a warning during startup is still a good idea.

Reply:
I am perfectly ok with **not** starting, unless the user demonstrates he knows what he is doing by setting the new parameter explicitly. I firmly believe this to be better tradeoff between our two "competing philosophies" on the matter.

Reply:
> I firmly believe this to be better tradeoff between our two "competing philosophies" on the matter.

You are right.  We still need to not accept the payload in the POST request because otherwise an attacker could still deploy a file on the agent (just not executing it).

Reply:
I agree. It'd rather complain early about incompatible options than deny something at runtime which would be harder for the confused operator to understand.